src/commands/compile.ts

```diff
@@ -189,36 +160,68 @@ export default class Compile extends Command {
           }
         }
         profiles.push({
-          path: resolvePath(dirname(superJsonPath), profileSettings.file),
+          path:
+            'file' in profileSettings
+              ? resolvePath(dirname(superJsonPath), profileSettings.file)
+              : undefined,
           maps,
-          id: ProfileId.fromId(flags.profileId, { userError }),
+          id: ProfileId.fromId(profile, { userError }),
         });
       }
     }
 
-    //Compile single profile and single map
-    if (flags.profileId && flags.providerName) {
+    //Compile single local profile and its local maps
+    if (flags.profileId && !flags.providerName) {
       const profileSettings = normalized.profiles[flags.profileId];
-      const profileProviderSettings =
-        profileSettings.providers[flags.providerName];
-      if ('file' in profileSettings) {
-        const maps: MapToCompile[] = [];
+      const maps: MapToCompile[] = [];
 
+      for (const [provider, profileProviderSettings] of Object.entries(
+        profileSettings.providers
+      )) {
         if ('file' in profileProviderSettings) {
           maps.push({
             path: resolvePath(
               dirname(superJsonPath),
               profileProviderSettings.file
             ),
-            provider: flags.providerName,
+            provider,
           });
         }
-        profiles.push({
-          path: resolvePath(dirname(superJsonPath), profileSettings.file),
-          maps,
-          id: ProfileId.fromId(flags.profileId, { userError }),
+      }
+      profiles.push({
+        path:
+          'file' in profileSettings
+            ? resolvePath(dirname(superJsonPath), profileSettings.file)
+            : undefined,
+        maps,
+        id: ProfileId.fromId(flags.profileId, { userError }),
+      });
```

Comment:
question: So this is main change, we are pushing remote profiles for compilation (which is later skipped) to be able to parse local maps. Because `compile` function expects hierarchy profile -> map. Maybe passing array of files to compile to `compile` function would be better structure?

Reply:
Yep, i was thinking that compile logic would take array of something like: {kind: 'map'; profileId: sting; provider: string; path: string} | {kind: 'profile'; profileId: sting;  path: string}

Reply:
Ok, i played with it a bit and if we go with : { kind: 'map', profileId: ProfileId; provider: string; path: string } | { kind: 'profile', profileId: ProfileId; path: string } as input we can significantly simplify the logic (we can use kind to access EXTENSIONS). But I am not doing this day before release so I will open issue and leave it after release.

Reply:
Thats exactly what I had in mind 👍 . And agree it is not right time to do it. I would just create an GH issue for it so it is tracked and can be done later.